mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoLocking.java

```diff
@@ -49,8 +51,24 @@
     private DBCollection coll;
     private long defaultTTL = 60l * 60l * 1000l;// 1 hr
 
+    // a set of locking collections which were already initialized
+    private static Set<String> initializedCollections = new HashSet<>();
+
     public MongoLocking(DBCollection coll) {
-        init(coll);
+        this(coll, false);
+    }
+
+    /**
+     *
+     * @param coll
+     * @param forceCollectionInit set to true in unit tests, where collections are wiped out between the tests
+     */
+    public MongoLocking(DBCollection coll, boolean forceCollectionInit) {
+        if (forceCollectionInit || !initializedCollections.contains(coll.getFullName())) {
+            init(coll);
+            initializedCollections.add(coll.getFullName());
```

Comment:
Collection name is not enough for the general case. Use dbname+collectionname. Also, synchronize the access to the set.